oslo/transformers/tasks/data_bart_pretraining.py

```diff
@@ -102,122 +142,149 @@ def __init__(
+        # determine how many tokens we need to mask in total
+        # masking something that is not a PAD token and not a Special token.
+        is_token_mask = ~(input_ids == self.tokenizer.pad_token_id) & ~special_tokens_mask_inputs
+        num_tokens_to_mask = int(math.ceil(is_token_mask.astype(float).sum() * self.mask_ratio))
+        if num_tokens_to_mask == 0:
+            return input_ids, labels
+
+        # generate a sufficient number of span lengths
+        span_lengths = np.random.poisson(lam=self.poisson_lambda, size=(num_tokens_to_mask,))
+        while np.cumsum(span_lengths, 0)[-1] < num_tokens_to_mask:
+            span_lengths = np.concatenate(
+                [span_lengths, np.random.poisson(lam=self.poisson_lambda, size=(num_tokens_to_mask,))]
+            )
+
+        # remove all spans of length 0
+        # note that BART inserts additional mask tokens where length == 0,
+        # which we do not implement for now as it adds additional complexity
+        span_lengths = span_lengths[span_lengths > 0]
+
+        # trim to about num_tokens_to_mask tokens
+        # Found index of span_length to select token as much as num_tokens_to_mask
+        cutoff_idx = np.argmin(np.abs(np.cumsum(span_lengths, 0) - num_tokens_to_mask)) + 1
+        span_lengths = span_lengths[:cutoff_idx]
+
+        # randomly choose starting positions for masking
+        token_indices = np.argwhere(is_token_mask )
+        span_starts = np.random.permutation(token_indices.shape[0])[: span_lengths.shape[0]]
+        # prepare mask
+        masked_indices = token_indices[span_starts]
+        mask = np.full_like(input_ids, fill_value=False)
+        
+        # masking process
+        # Put a True value in the index corresponding to the masked_indices, 
+        # do -1 in span_length since one mask token is in it, and calculate remaining whether there is anything left to mask.
+        # The index of masked_indices is +1 to point to the next token.
+        # Repeat the above process to insert the mask token until the span_length is all 0 or less and the masked_indices are less than max_index.
+        
+        # mask starting positions
+        for mi in masked_indices:
+            mask[tuple(mi)] = True
+        span_lengths -= 1
+        # fill up spans
+        max_index = input_ids.shape[1] - 1
+        remaining = (span_lengths > 0) & (masked_indices[:, 1] < max_index)
+        while np.any(remaining):
+            # Need to modify only the mainnig part repeatedly without repeating all masked_indices, B.S
+            masked_indices[remaining, 1] += 1
+            for mi in masked_indices:
+                mask[tuple(mi)] = True
+            span_lengths -= 1
+            remaining = (span_lengths > 0) & (masked_indices[:, 1] < max_index)
+
+        # place the mask tokens
+        # If the mask corresponds to special_tokens_mask_inputs, enter False because you should not mask
+        mask[np.where(special_tokens_mask_inputs)] = False
```

Comment:
Issue
Line 277 ~ 280 why need?
I want to talk about this.

Should the special token always be removed from the label?